include/seqan3/range/view/persist.hpp

```diff
@@ -87,7 +87,7 @@ class view_persist : public ranges::view_interface<view_persist<urng_t>>
     //!\brief The value_type (which equals the reference_type with any references removed).
     using value_type        = value_type_t<urng_t>;
     //!\brief The size_type is `size_t` if the the view is exact, otherwise void.
-    using size_type         = std::conditional_t<std::ranges::SizedRange<urng_t>, size_type_t<urng_t>, void>;
+    using size_type         = typename detail::size_type_t_or_void<urng_t>::type;
```

Comment:
Do we need `detail::size_type_t_or_void<urng_t>::type` a lot in the near future? I found the previous line okay and I think we could live without this rather specific meta function.

Reply:
> I found the previous line okay

I did, too, but the compiler unfortunately didn't! The problem was that for non sized ranges, the metafunction `size_type_t` is not defined at all and this would lead to an error. I needed to add the other metafunction to work around this, I wouldn't have done it otherwise :expressionless: